src/qibo/backends/__init__.py

```diff
@@ -138,12 +138,14 @@ def _default_transpiler(cls):
         qubits = cls._backend.qubits
         natives = cls._backend.natives
         connectivity_edges = cls._backend.connectivity
-        if (
-            qubits is not None
-            and natives is not None
-            and connectivity_edges is not None
-        ):
-            connectivity = nx.Graph(connectivity_edges)
+        if qubits is not None and natives is not None:
+            if len(connectivity_edges) == 0:
+                connectivity = nx.Graph()
+                qubits = cls._backend.platform.qubits.keys()
+                connectivity.add_nodes_from(qubits)
+            else:
+                connectivity = nx.Graph(connectivity_edges)
```

Comment:
Ah yes, this is partially my fault, since I told Changsoo that `connectivity_edges` would have added the nodes automatically...
I did not foresee this case :sweat_smile: 

Maybe this is enough:
```suggestion
            connectivity = nx.Graph(connectivity_edges) if connectivity_edges is not None else nx.Graph()
            connectivity.add_nodes_from(qubits)
```
- always adding `qubits` should not be a problem in general
- `qubits` should be read from the backend, not from the `platform` - which is a Qibolab detail, but it's not needed because `Backend` standardized the interface